### PR TITLE
Haproxy upgrade (epicli)

### PR DESCRIPTION
**Pull request by sk4zuzu**
#1586 should be merged first, then this one rebased and reviewed :+1:

### CODE REVIEW

File: core/src/epicli/cli/engine/ansible/AnsibleVarsGenerator.py [hunk 1]
```diff
@@ -53,38 +55,64 @@ def generate(self):
...
 
         with open(vars_file_path, 'w') as stream:
             dump(document, stream)
 
-        self.roles_with_generated_vars.append(to_role_name(role))
+        if vars_file_name == 'main.yml':
+            self.roles_with_generated_vars.append(to_role_name(role))
+
+    def write_role_manifest_vars(self, ansible_dir, role, kind):
+        enabled_kinds = {"configuration/haproxy"}
```
Comment (atsikham) [inline core/src/epicli/cli/engine/ansible/AnsibleVarsGenerator.py:94]: Why dict is used here and not tuple for example?

Comment (atsikham) [reply core/src/epicli/cli/engine/ansible/AnsibleVarsGenerator.py:94]: Could you explain why do we need both `enabled_roles` and `enabled_kinds`?

**Review by atsikham — CHANGES_REQUESTED** [review]
LGTM, but with a few questions, will be approved then

File: core/src/epicli/cli/engine/ansible/AnsibleVarsGenerator.py (hunk 1 above)
Comment (sk4zuzu) [reply core/src/epicli/cli/engine/ansible/AnsibleVarsGenerator.py:94]: > Why dict is used here and not tuple for example?

This is a set(), not a dict(). In python {} have double meaning, depending on syntax of the literal.

Comment (sk4zuzu) [reply core/src/epicli/cli/engine/ansible/AnsibleVarsGenerator.py:94]: > Could you explain why do we need both `enabled_roles` and `enabled_kinds`?

This is just an optimization. This is a new thing never done before, so after some discussion with @to-bar we decided to narrow the scope of this operation. Default behavior is still to use default values.

Comment (atsikham) [reply core/src/epicli/cli/engine/ansible/AnsibleVarsGenerator.py:94]: > > Why dict is used here and not tuple for example?
> 
> This is a set(), not a dict(). In python {} have double meaning, depending on syntax of the literal.

Agree, it's my inattention

**Review by atsikham — APPROVED** [review]

**Review by atsikham — APPROVED** [review]

**Review by rafzei — APPROVED** [review]
LGTM but I see no docs update (components,readme etc..)

**Comment by sk4zuzu** [comment]
> LGTM but I see no docs update (components,changelog etc..)

@rafzei, please re-check now. I added components, changelog and some info how to expose k8s service (it was TODO previously).

**Review by rafzei — APPROVED** [review]
Nice, thanks.

**Review by atsikham — APPROVED** [review]

**Pull request merged**